src/unix/fs.c

```diff
@@ -1240,9 +1241,28 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
   return r;
 }
 
-static ssize_t uv__fs_copyfile(uv_fs_t* req) {
 #if defined(__APPLE__) && !TARGET_OS_IPHONE
-  /* On macOS, use the native copyfile(3). */
+
+static ssize_t uv__fs_clonefile_mac(uv_fs_t* req) {
+  static int has_clone_acl = 1;
+
+  /* Note: clonefile() does not set the group ID on the destination
+   * file correctly. */
+  if (!clonefile(req->path, req->new_path, has_clone_acl ? UV__FS_CLONE_ACL : 0))
+    return 0;  /* success */
+
+  if (errno == EINVAL) {
+    has_clone_acl = 0;
+    errno = 0;
+    /* UV__FS_CLONE_ACL flag not supported (macOS < 13); try without. */
+    if (!clonefile(req->path, req->new_path, 0))
+      return 0;  /* success */
+  }
+
+  return 1;
```

Comment:
```suggestion
  return UV__ERR(errno);
```
And maybe change the return type from `ssize_t` to `int` because a casual reader probably expects it to return the number of bytes copied.

(I realize you copied that from `uv__fs_copyfile()`. You can change that function's return type as well if you want.)